File: Source/Code/CorePlugin/Brick.cs
using System;
using Breakout.FSharp;
using Duality;
using Duality.Resources;

namespace Breakout
{
	[Serializable]
	public class BrickOld : Component, ICmpCollisionListener
	{
		public void OnCollisionBegin(Component sender, CollisionEventArgs args)
		{
			var scoreComponent = Scene.Current.FindComponent<ScoreComponent>();
			if (scoreComponent != null)
				scoreComponent.IncreaseScore(1);
			GameObj.DisposeLater();
		}

		public void OnCollisionEnd(Component sender, CollisionEventArgs args)
		{
			
		}

		public void OnCollisionSolve(Component sender, CollisionEventArgs args)
		{
			
		}
	}
}

File: Source/Code/Breakout.CorePlugin/Brick.fs
namespace Breakout

open Duality
open Duality.Resources
open System
open Duality.Components.Renderers
open Breakout.FSharp

[<Serializable>]
type Brick() = 
    inherit Component()   
    
    interface ICmpCollisionListener with 
        member this.OnCollisionBegin (_, _ )=            
            let scoreComponent = Scene.Current.FindComponent<ScoreComponentF>()
            //if (scoreComponent <> null) then
            scoreComponent.IncreaseScore 1 
            this.GameObj.DisposeLater();
        member this.OnCollisionEnd(_,_)=  
            ()
        member this.OnCollisionSolve(_,_)=  
            ()